wandb/apis/public/utils.py

```
@@ -103,3 +113,33 @@ def fetch_org_from_settings_or_entity(
         )
         organization = entity_org.display_name
     return organization
+
+
+@lru_cache(maxsize=128)
+def check_server_feature(client: "Client", feature: ServerFeature) -> bool:
+    """Check if a server feature is enabled.
+
+    Args:
+        feature (ServerFeature): The feature to check.
+
+    Returns:
+        bool: True if the feature is enabled, False otherwise.
+    """
+    try:
+        response = client.execute(gql(SERVER_FEATURES_QUERY_GQL))
+    except Exception as e:
+        if 'Cannot query field "features" on type "ServerInfo".' in str(e):
+            wandb.termwarn(
+                "Possible that the server doesn't support the feature query, please make sure you are on an updated server release version"
+            )
+            return False
```

Should we warn here? Since we will throw a `SystemError` when returning false.

---

i don't think we should throw an error if its the case that their server version is so old that they don't have the server `features` field? I think it should just return false? Because for other featureflag checks in the future an api might just want to see if they want to do certain logic depending on if its an old server instead of erroring out

---

I agree that we should return `False` in that case.
Mainly I am wondering/arguing if we should do `wandb.termwarn` and instead leave logging that information up to each caller. 

My thought is that there might be a case where we would want to transparently handle a fallback case when the server does not support a feature.

---

I ended up putting a wrapper around this function and this function errors out whereas the wrapper returns false and warns. this way users can pick which one they want to use